common/src/main/java/org/apache/uniffle/common/netty/MessageEncoder.java

```diff
@@ -59,5 +59,9 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       byteBuf.release();
     }
     ctx.writeAndFlush(byteBuf);
+    // do transferTo send data after encode buffer send.
+    if (message instanceof Transferable) {
+      ((Transferable) message).transferTo(ctx.channel());
```

Comment:
@jerqi the line is write the FileSegmentManageBuffer ShuffleData.

Reply:
OK.